src/ReactLeafletMap/index.js

```diff
@@ -87,20 +87,29 @@ class ReactLeafletMap extends React.PureComponent {
     }
 
     onViewportChange(viewport) {
-        const center = {
-            lat: viewport.center[0],
-            lon: viewport.center[1]
-        };
-
-        const boxRange = viewUtils.getBoxRangeFromZoomLevel(viewport.zoom);
+        const change = {};
+        // viewport.center could be undefined
+        if(viewport && viewport.hasOwnProperty('center') && viewport.center) {
+            const center = {
+                lat: viewport.center[0],
+                lon: viewport.center[1]
+            };
+            change['center'] = center;
+        }
+        
+        // viewport.zoom could be undefined
+        if(viewport && viewport.hasOwnProperty('zoom') && Number.isFinite(viewport.zoom)) {
+            const boxRange = viewUtils.getBoxRangeFromZoomLevel(viewport.zoom);
+            change['boxRange'] = boxRange;
+        }
 
         // TODO for IndexedVectorLayer rerender (see IndexedVectorLayer render method)
         let stateUpdate = {viewport};
 
         if (this.props.onViewChange) {
```

Comment:
Test in change is not epmty for whole if/else?

Reply:
You mean dont call onViewChange if change is empty?

Reply:
From react-leaflet doc: onViewportChange: (viewport: {center: ?[number, number], zoom: ?number}) => void (optional): fired continuously as the viewport changes.

So every onViewportChange contains center or zoom change.

Reply:
props.onViewChange and onViewportChange are two different methods